data-resource/src/blake3.rs

```diff
@@ -0,0 +1,84 @@
+use crate::ResourceIdTrait;
+use data_error::Result;
+use serde::{Deserialize, Serialize};
+use std::{
+    fs,
+    io::{BufRead, BufReader},
+    path::Path,
+};
+
+use blake3::Hasher;
+use hex::encode;
+
+/// Represents a resource identifier using the BLAKE3 algorithm.
+///
+/// Uses [`blake3`] crate to compute the hash value.
+#[derive(
+    Eq,
+    Ord,
+    PartialEq,
+    PartialOrd,
+    Hash,
+    Clone,
+    Copy,
+    Debug,
+    Deserialize,
+    Serialize,
+)]
+pub struct ResourceId {}
```

Comment:
Is it stub for future implementation? We have `"cryptographic-hash"` enabled by default, so we should provide the default implementation.

Reply:
The `ResourceId` struct doesn't hold any data because there is no need to. Instead, methods like `from_path` and `from_bytes` return hash values directly. 

The `ResourceIdTrait` simply enforces some restrictions on types implementing it, to allow us to use it from other crates without worrying about the underlying hash function.